tests/nnstreamer_filter_snpe/unittest_filter_snpe.cc

```diff
@@ -0,0 +1,362 @@
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("snpe");
+  EXPECT_NE (sp, nullptr);
+  _SetFilterProp (&prop, "snpe", model_files);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+
+  GstTensorsInfo in_info, out_info;
+
+  ret = sp->getModelInfo (NULL, NULL, data, GET_IN_OUT_INFO, &in_info, &out_info);
+  EXPECT_EQ (ret, 0);
+
+  EXPECT_EQ (in_info.num_tensors, 1U);
+  EXPECT_EQ (in_info.info[0].dimension[0], 1U);
+  EXPECT_EQ (in_info.info[0].dimension[1], 1U);
+  EXPECT_EQ (in_info.info[0].dimension[2], 1U);
+  EXPECT_EQ (in_info.info[0].dimension[3], 1U);
+  EXPECT_EQ (in_info.info[0].type, _NNS_FLOAT32);
+
+  EXPECT_EQ (out_info.num_tensors, 1U);
+  EXPECT_EQ (out_info.info[0].dimension[0], 1U);
+  EXPECT_EQ (out_info.info[0].dimension[1], 1U);
+  EXPECT_EQ (out_info.info[0].dimension[2], 1U);
+  EXPECT_EQ (out_info.info[0].dimension[3], 1U);
+  EXPECT_EQ (out_info.info[0].type, _NNS_FLOAT32);
+
+  sp->close (&prop, &data);
+  g_free (model_file);
+}
+
+/**
+ * @brief Negative case calling getModelInfo before open
+ */
+TEST (nnstreamerFilterSnpe, getModelInfo01_n)
+{
+  int ret;
+  void *data = NULL;
+  gchar *model_file;
+  GstTensorFilterProperties prop;
+ASSERT_TRUE (_GetModelFilePath (&model_file));
+
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("snpe");
+  EXPECT_NE (sp, nullptr);
+
+  GstTensorsInfo in_info, out_info;
+
+  ret = sp->getModelInfo (NULL, NULL, data, SET_INPUT_INFO, &in_info, &out_info);
+  EXPECT_NE (ret, 0);
+  _SetFilterProp (&prop, "snpe", model_files);
+
+  sp->close (&prop, &data);
+  g_free (model_file);
+}
+
+/**
+ * @brief Negative case with invalid argument
+ */
+TEST (nnstreamerFilterSnpe, getModelInfo02_n)
+{
+  int ret;
+  void *data = NULL;
+  gchar *model_file;
+  GstTensorFilterProperties prop;
+  ASSERT_TRUE (_GetModelFilePath (&model_file));
+
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("snpe");
+  EXPECT_NE (sp, nullptr);
+  _SetFilterProp (&prop, "snpe", model_files);
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+  sp->close (&prop, &data);
+
+  GstTensorsInfo in_info, out_info;
+
+  /* not supported */
+  ret = sp->getModelInfo (NULL, NULL, data, SET_INPUT_INFO, &in_info, &out_info);
+  EXPECT_NE (ret, 0);
+
+  sp->close (&prop, &data);
+  g_free (model_file);
+}
+
+/**
+ * @brief Test snpe subplugin with successful invoke for sample dlc model
+ */
+TEST (nnstreamerFilterSnpe, invoke00)
+{
+  int ret;
+  void *data = NULL;
+  GstTensorMemory input, output;
+  gchar *model_file;
+  GstTensorFilterProperties prop;
+
+  ASSERT_TRUE (_GetModelFilePath (&model_file));
+  const gchar *model_files[] = {
+    model_file,
+    NULL,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("snpe");
+  EXPECT_NE (sp, (void *) NULL);
```

Comment:
ASSERT_TRUE (sp != nullptr); as sp is null, this will throw exception.

Reply:
Thanks, if the pointer is null, further tests are meaningless. I updated the nullptr check with `ASSERT_TRUE`. PTAL